### PR TITLE
System Check Script #7: Sun Sensors 

### PR DESCRIPTION
**Pull request by yashika-batra**
Sun sensor test: clean up + style/documentation fixes

### CODE REVIEW

File: state_machine/applications/system-check/tests/sun_sensor_test.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    elif sun_sensor_index == "-X":
+        return cubesat._cubesat.sun_xn.lux
+    elif sun_sensor_index == "+Y":
+        return cubesat._cubesat.sun_yp.lux
+    elif sun_sensor_index == "+Z":
+        return cubesat._cubesat.sun_zp.lux
+    elif sun_sensor_index == "+X":
+        return cubesat._cubesat.sun_xp.lux
+    else:
+        print(sun_sensor_index, "is not a defined sun sensor.")
```
Comment (jbwillis) [inline state_machine/applications/system-check/tests/sun_sensor_test.py:27]: I would rather see this raise a `ValueError` than just print something so we get the full stacktrace.

File: state_machine/applications/system-check/tests/sun_sensor_test.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+              "Waiting {wait_time} seconds.")
+
+    time.sleep(wait_time)
+    print("Collecting sensor data...")
+    lux_val = lux(sensor_index)
+    print("Data collection complete.")
+    return lux_val
+
+
+def test_sensorx(result_dict, sensor_index, light_dark):
```
Comment (jbwillis) [inline state_machine/applications/system-check/tests/sun_sensor_test.py:54]: don't love the name of this function (not very descriptive). How about `test_individual_sun_sensor`?

File: state_machine/applications/system-check/tests/sun_sensor_test.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    Process user test results and update the result dictionary accordingly
+    """
+
+    # get lux value from the user test
+    lux_val = user_test(sensor_index, light_dark)
+
+    # find result key and result value strings
+    result_key = 'Sun' + sensor_index + '_' + light_dark
+    if lux_val:
+        result_val_string = ("Testing Sun Sensor {sensor_index} in the" +
```
Comment (jbwillis) [inline state_machine/applications/system-check/tests/sun_sensor_test.py:66]: This should be an f string (missing the `f`)

**Review by jbwillis — CHANGES_REQUESTED** [review]

File: state_machine/applications/system-check/tests/sun_sensor_test.py [hunk 4]
```diff
@@ -25,7 +24,7 @@ def lux(sun_sensor_index):
         return cubesat._cubesat.sun_xp.lux
     else:
         print(sun_sensor_index, "is not a defined sun sensor.")
-        return
+        raise ValueError
```
Comment (jbwillis) [inline state_machine/applications/system-check/tests/sun_sensor_test.py:27]: `raise ValueError(f"Sun sensor index {sun_sensor_index} is not defined")`

Comment (jbwillis) [reply state_machine/applications/system-check/tests/sun_sensor_test.py:27]: You can put the error message in the error rather than printing it here.

**Review by jbwillis — CHANGES_REQUESTED** [review]

File: state_machine/applications/system-check/tests/sun_sensor_test.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    # else continue
+    print("Collecting sensor data...")
+    lux_val = lux(sensor_index)
+    print("Data collection complete.")
+    return lux_val
+
+
+def test_individual_sun_sensor(result_dict, sensor_index, light_dark):
+    """
+    All automation happens in this function
```
Comment (jbwillis) [inline state_machine/applications/system-check/tests/sun_sensor_test.py:57]: Describe the test here. What is being done?

**Review by jbwillis — CHANGES_REQUESTED** [review]

File: state_machine/applications/system-check/tests/sun_sensor_test.py (hunk 4 above)
Comment (yashika-batra) [reply state_machine/applications/system-check/tests/sun_sensor_test.py:27]: Thanks!

**Review by jbwillis — APPROVED** [review]

**Pull request merged**